lab_1/main.py

```diff
@@ -1,128 +1,175 @@
-"""
-Lab 1
-A concordance extraction
-"""
-
 
 def tokenize(text: str) -> list:
-    """
-    Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
-    :param text: the initial text
-    :return: a list of lowercased tokens without punctuation
-    e.g. text = 'The weather is sunny, the man is happy.'
-    --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    """
-    pass
-
+    if not isinstance(text, str):
+        return []
+
+    clean_tokens = []
+    for token in text.lower().split():
+        word = ''
+        for character in token:
+            if character.isalpha():
+                word += character
+            if word:
+                clean_tokens.append(word)
```

Comment:
It should not be nested, you should append the `word` variable in the outer `for`-loop.